src/segment-loader.js

```diff
@@ -1104,4 +1109,42 @@ export default class SegmentLoader extends videojs.EventTarget {
     this.throughput.rate +=
       (segmentProcessingThroughput - rate) / (++this.throughput.count);
   }
+
+  /**
+   * Adds a cue to the segment-metadata track with some metadata information about the
+   * segment
+   *
+   * @private
+   * @param {Object} segmentInfo
+   *        the object returned by loadSegment
+   * @method addSegmentMetadataCue_
+   */
+  addSegmentMetadataCue_(segmentInfo) {
+    if (!this.segmentMetadataTrack_ || !segmentInfo) {
+      return;
+    }
+
+    const playlist = segmentInfo.playlist;
+    const segment = segmentInfo.segment;
+    const start = segment.start;
+    const end = segment.end;
+
+    removeCuesFromTrack(start, end, this.segmentMetadataTrack_);
+
+    const value = {
+      uri: segmentInfo.uri,
+      timeline: segmentInfo.timeline,
+      playlist: playlist.uri,
+      start,
+      end
+    };
+    const Cue = window.WebKitDataCue || window.VTTCue;
+    const data = JSON.stringify(value);
+    const cue = new Cue(start, end, data);
+
+    value.data = data;
+    cue.value = value;
```

Comment:
Yes and no. It mainly depends on the `const Cue = window.WebKitDataCue || window.VTTCue;` line. I was following what is done in contrib-media-sources when adding id3 tags, but `WebKitDataCue` is only in safari right now, and there are some small differences between `WebKitDataCue` and `VTTCue`. Mainly, `WebKitDataCue` does not have a `text` property but instead a `value`. For id3 tags this `value` property has a `key` and `data` property usually. My goal with these lines was to minimize the differences between the two cues so that you can just use `cue.value` in all the browsers. After having to explain this, it may just be better to only use `VTTCue` and not jump through all these hoops